keras_nlp/models/preprocessor.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tensorflow import keras
+
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class Preprocessor(keras.layers.Layer):
+    def __init__(
```

Comment:
does this actually do anything? why not remove this?

Reply:
Isn't it useful for when people pass args like `name`?

Reply:
What is the default behavior without init? I thought we always had to chain to super....

Reply:
Without `__init__`, it won't call `keras.layers.Layer`'s constructor. That's why I mentioned `name` above

Reply:
Hmmm, to the contrary...

```
>>> class A:
...   def __init__(self):
...     print("hello from A")
... 
>>> class B(A):
...   pass
... 
>>> class C(B):
...   def __init__(self):
...     super().__init__()
... 
>>> c = C()
hello from A
```



```
>>> class A:
...   def __init__(self):
...     print("hello from A")
... 
>>> class B(A):
...   def __init__(self):
...       pass
... 
>>> class C(B):
...   def __init__(self):
...     super().__init__()
... 
>>> c = C()
>>> 
```

Interesting. So, I guess if we don't override the constructor in class B, class B's constructor is class A's constructor. I now remember my OOP prof mentioning this, lol

Reply:
Yeah, if all you do is pass all arguments up the change to super, you don't need this. That is already the default behavior.